onnxruntime/python/tools/transformers/gpt2_beamsearch_helper.py

```diff
@@ -0,0 +1,822 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation.  All rights reserved.
+# Licensed under the MIT License.  See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# This script helps onnx conversion and validation for GPT2 model with past state.
+import os
+import logging
+import torch
+import onnx
+import random
+import numpy
+import time
+import re
+from pathlib import Path
+from typing import List, Dict, Tuple, Union
+from transformers import GPT2LMHeadModel, GPT2Config
+from benchmark_helper import Precision
+from gpt2_helper import Gpt2Helper, Gpt2Inputs, GPT2ModelNoPastState, MyGPT2Model, MyGPT2LMHeadModel, MyGPT2LMHeadModel_NoPadding
+
+logger = logging.getLogger(__name__)
+
+class Gpt2HelperFactory:
+    @staticmethod
+    def create_helper(helper_type="default"):
+        helpers = {
+            "default": Gpt2Helper,
+            "beam_search_step": Gpt2BeamSearchHelper,
+        }
+        w = helpers[helper_type]
+        return w
+
+class GPT2LMHeadModel_BeamSearchStep(GPT2LMHeadModel):
+    """Here we wrap a class for Onnx model conversion for GPT2LMHeadModel with past state."""
```

Comment:
It is better to indicate that it includes one step beam search.

Reply:
Thanks for the suggestion. Added in the update.